pyramid/predicates.py

```diff
@@ -132,14 +132,19 @@ def __call__(self, context, request):
 class AcceptPredicate(object):
     def __init__(self, val, config):
         self.val = val
+        if '*' in self.val:
+            raise ConfigurationError(
+                '"accept" predicate does not accept media ranges, only '
+                'specific media types',
+            )
```

Comment:
Might it be better for the `ConfigurationError` to be raised within `add_view()` and `add_route`? It'd happen earlier; plus as mentioned above, the call to the predicate in view matching seems (if I haven't missed something) unnecessary and duplicates the work of the call to `best_match`/the first call to `acceptable_offers` that treats it as a non-predicate argument?

Reply:
The reason that I have done it here is that otherwise the check would need to be done in both `add_route` and `add_view` at the very least and this will raise at config-time anyway which is the primary goal here.